### PR TITLE
708 Datapoints

### PR DESCRIPTION
**Pull request by TeeJayYang**

### CODE REVIEW

File: src/cea708.c [hunk 1]
```diff
@@ -64,6 +66,10 @@ void cea708_parse_user_data_type_structure(const uint8_t* data, size_t size, use
     user_data->em_data = data[1];
     data += 2, size -= 2;
 
+    if (detail && size != user_data->cc_count * 3 + 1){
```
Comment (TeeJayYang) [inline src/cea708.c]: This needs to be changed to `detail && size/3 != user_data->cc_count`

File: src/caption.c [hunk 2]
```diff
@@ -422,6 +424,226 @@ libcaption_status_t caption_frame_decode(caption_frame_t* frame, uint16_t cc_dat
...
+int c1_code_length[32] = { 1, 1, 1, 1, 1, 1, 1, 1,
+                           2, 2, 2, 2, 2, 2, 1, 1,
+                           3, 4, 3, 1, 1, 1, 1, 5,
+                           7, 7, 7, 7, 7, 7, 7, 7};
+
+libcaption_status_t caption_frame_decode_dtvcc(caption_frame_t* frame, uint16_t cc_data, 
+                                               double timestamp, cea708_cc_type_t type){
+    dtvcc_packet_t *packet = &frame->state.dtvcc_packet;
+    if (cc_type_dtvcc_packet_header == type) {
+        int current_sequence_number = cc_data >> 14;
```
Comment (colekas-imax) [inline src/caption.c]: What width is current_sequence_number? We should try to use the smallest possible (and mask the assignment to ensure the results).

File: src/caption.c [hunk 3]
```diff
@@ -422,6 +424,226 @@ libcaption_status_t caption_frame_decode(caption_frame_t* frame, uint16_t cc_dat
...
+                ++(frame->detail.packetLoss);
+            }
+            packet->sequence_count %= 4;
+            packet->seen_sequences = 0;
+        }
+
+        packet->sequence_number = current_sequence_number;
+        packet->packet_size = (cc_data >> 8) & 0x2f;
+        packet->service_number = (cc_data & 0xe0) >> 5;
+        packet->block_size =  cc_data & 0x1f;
```
Comment (colekas-imax) [inline src/caption.c]: We should be consistent with the ordering of shifts and ands

File: src/caption.c [hunk 4]
```diff
@@ -422,6 +424,226 @@ libcaption_status_t caption_frame_decode(caption_frame_t* frame, uint16_t cc_dat
...
+        if (!(packet->block_size <= 31)){
+            status_detail_set(&frame->detail, LIBCAPTION_DETAIL_ABNORMAL_SERVICE_BLOCK);
+        }
```
Comment (colekas-imax) [inline src/caption.c:460]: if we're and'ing cc_data with 0x1F, then the max value will always be 31 so the above check will always be false. Is this similar to what check that was in FFmpeg?

File: src/cea708.c [hunk 5]
```diff
@@ -64,6 +66,10 @@ void cea708_parse_user_data_type_structure(const uint8_t* data, size_t size, use
     user_data->em_data = data[1];
     data += 2, size -= 2;
 
+    if (detail && size/3 != user_data->cc_count){
```
Comment (colekas-imax) [inline src/cea708.c:69]: Why a third? Did you mean to right shift by 3?

**Review by colekas-ssim — DISMISSED** [review]

File: src/caption.c (hunk 4 above)
Comment (TeeJayYang) [reply src/caption.c:460]: The same check is made in FFmpeg

File: src/cea708.c (hunk 5 above)
Comment (TeeJayYang) [reply src/cea708.c:69]: cc_count is the number of sections of data (each section has the data itself and an extra metadata byte). This totals 3 bytes per section and size is the number of bytes in total.

**Review by colekas-ssim — APPROVED** [review]

**Pull request merged**